api/types/swarm/network.go

```diff
@@ -78,12 +78,14 @@ type Network struct {
 // NetworkSpec represents the spec of a network.
 type NetworkSpec struct {
 	Annotations
-	DriverConfiguration *Driver      `json:",omitempty"`
-	IPv6Enabled         bool         `json:",omitempty"`
-	Internal            bool         `json:",omitempty"`
-	Attachable          bool         `json:",omitempty"`
-	Ingress             bool         `json:",omitempty"`
-	IPAMOptions         *IPAMOptions `json:",omitempty"`
+	DriverConfiguration *Driver           `json:",omitempty"`
+	IPv6Enabled         bool              `json:",omitempty"`
+	Internal            bool              `json:",omitempty"`
+	Attachable          bool              `json:",omitempty"`
+	Ingress             bool              `json:",omitempty"`
+	IPAMOptions         *IPAMOptions      `json:",omitempty"`
+	Reference           *NetworkReference `json:",omitempty"`
```

Comment:
In order to address your very valid concern about api extendability, or even to be able to drop this functionality  in future, I am thinking the structure we put in here to have another field (besides the `Network` or maybe  better call it  generically `Target`) `operation` which says what the target is for.

So that 

```
type Reference struct {
    Target string
    Operation string // or enum type
}
```

where operation would have now only one value "config"

WDYT?